test/compression.js

```diff
@@ -676,6 +676,35 @@ describe('compression()', function () {
       .end()
     })
   })
+
+  describe('when callbacks are used', function () {
+    it('should call the passed callbacks', function(done){
+      var callbacks = 0;
+      var server = createServer({ threshold: '1kb' }, function (req, res) {
+        res.setHeader('Content-Type', 'text/plain')
+        res.write('Hello', null, function(){
+          callbacks++
+          res.flush()
+        });
+        res.write(' World', null, function(){
+          callbacks++
+          res.flush()
+        });
+        res.end(null, null, function(){
+          callbacks++
+        });
+      })
+
+      request(server)
+      .get('/')
+      .set('Accept-Encoding', 'gzip')
+      .expect('Content-Encoding', 'gzip')
+      .end(function(){
```

Comment:
The first argument to this callback is an `err` and the code here is not handling the error. Please add error handling.